src/soter/openssl/soter_bignum_utils.h

```diff
@@ -68,6 +71,21 @@ static void memcpy_big_endian(void* dst, const void* src, size_t size)
     }
 }
 #endif
+
+static int get_bn_param(
+    const EVP_PKEY* pkey, const char* name, unsigned char* buf, size_t buf_size, BIGNUM** bn)
+{
+    OSSL_PARAM params[2];
+
+    params[0] = OSSL_PARAM_construct_BN(name, buf, buf_size);
+    params[1] = OSSL_PARAM_construct_end();
+
+    if (!EVP_PKEY_get_params(pkey, params)) {
+        return 0;
+    }
+
+    return OSSL_PARAM_get_BN(params, bn);
```

Comment:
As I understand from docs and what makes sense for me, the `OSSL_PARAM_get_BN` returns the bignum from one `OSSL_PARAM`, and not an array of `OSSL_PARAM`s, though it accepts a pointer. If it's true, I would prefer an explicit syntax, something like
```c
return OSSL_PARAM_get_BN(&params[0], bn);
```

just to emphasize that we process the first element and not the entire array.